From de5451f44cac7b91ac39af33c511ca9ed4d640c9 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Thu, 30 Jun 2022 09:16:58 +0800
Subject: [PATCH 010/100] backend-drm: Don't try to prepare non-cursor view in
 renderer-only mode

Return before attempting to create drm_fb, since it would fail out in
the later check anyway.

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 libweston/backend-drm/state-propose.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/libweston/backend-drm/state-propose.c b/libweston/backend-drm/state-propose.c
index 18a6d628be..c55a9aa0e5 100644
--- a/libweston/backend-drm/state-propose.c
+++ b/libweston/backend-drm/state-propose.c
@@ -404,6 +404,11 @@ drm_output_find_plane_for_view(struct drm_output_state *state,
 		return NULL;
 	}
 
+	/* filter out non-cursor views in renderer-only mode */
+	if (mode == DRM_OUTPUT_PROPOSE_STATE_RENDERER_ONLY &&
+	    ev->layer_link.layer != &b->compositor->cursor_layer)
+			return NULL;
+
 	buffer = ev->surface->buffer_ref.buffer;
 	if (buffer->type == WESTON_BUFFER_SOLID) {
 		pnode->try_view_on_plane_failure_reasons |=
-- 
2.20.1

